### PR TITLE
Sillier EMAG effects for the lawbringer

### PR DESCRIPTION
**Pull request by MeggalBozale**
[Feature] [Game-Objects]
## About the PR 
Adds a more fitting set of emag effects for the lawbringer.
- When you shoot the lawbringer while it's emagged, it will pick between random modes for you.
- ~~There is a 1% chance to force you to scream out "I AM THE LAW" and trigger the explosion~~
- Emagged lawbringers have no sense of laws, and so the HoS is not immune to the explosion for saying these words.
- Moves over switching code to a proc so that it can be accessed from other places in the code. There's a lot of custom varediting in there, so it should be exposed to more parts of the code for use. Like this!

## Why's this needed? 

https://github.com/goonstation/goonstation/assets/27376947/be5eff0f-2313-478e-9a43-c7cc8dc8bb65

NOTE: VIDEO SHOWS THE CHANCE TO BE FORCED TO SAY "I AM THE LAW". NO LONGER A FEATURE.

```changelog
(u)MeggalBozale
(+)The emag effect for the lawbringer has been modified slightly.
```

### CODE REVIEW

**Comment by MeggalBozale** [comment]
Note that while prob(1) is prob(1) i cut out like 30 seconds of me spamming it to trigger the IATL effect

**Comment by MeggalBozale** [comment]
ALSO IMPORTANT TO CLARIFY I FORGOT!!!
this doesnt stop you from getting access to the lawbringer yourself, current emag effect isnt removed :)

**Comment by TheRyeGuyWhoWillNowDie** [comment]
seems not fun? now even if you kill the head of security, steal their gun, and have an emag, the lawbringer is little more than a toy, and worse yet extended use will get you gibbed

**Comment by MeggalBozale** [comment]
You can fabricate a new one or just not emag the thing. It's an emag! Weird things can happen!

File: code/obj/item/gun/energy.dm [hunk 1]
```diff
@@ -1512,6 +1524,14 @@ TYPEINFO(/obj/item/gun/energy/lawbringer)
 		return 0
 
 	shoot(turf/target, turf/start, mob/user, POX, POY, is_dual_wield, atom/called_target = null)
+
+		if (src.emagged)
+			if (prob(1))
+				boutput(user, SPAN_ALERT("You let it slip."))
+				user.say("I AM THE LAW")
+				src.are_you_the_law(user, "I AM THE LAW") // the thoughts won
```
Comment (TobleroneSwordfish) [inline code/obj/item/gun/energy.dm:1532]: Funny but probably a bad idea

**Review by TobleroneSwordfish — CHANGES_REQUESTED** [review]

File: code/obj/item/gun/energy.dm (hunk 1 above)
Comment (MeggalBozale) [reply code/obj/item/gun/energy.dm:1532]: fine :(

**Review by TobleroneSwordfish — APPROVED** [review]

**Review by Glamurio — APPROVED** [review]

**Pull request merged**